include/pro/pro.hpp

```diff
@@ -89,6 +94,70 @@ class ProWrapper : public Napi::ObjectWrap<ProWrapper> {
             return obj;
         });
     };
+
+    static Napi::Value proProofRequestBody(const Napi::CallbackInfo& info) {
+        return wrapResult(info, [&] {
+            // we expect arguments that match:
+            // first: {
+            //   "version": string,
+            //   "master_privkey": Uint8Array,
+            //   "rotating_privkey": Uint8Array,
+            //   "unix_ts": number,
+            // }
+
+            assertInfoLength(info, 1);
+            assertIsObject(info[0]);
+            auto env = info.Env();
+
+            auto first = info[0].As<Napi::Object>();
+
+            if (first.IsEmpty())
+                throw std::invalid_argument("proProofRequestBody first received empty");
+
+            assertIsNumber(first.Get("version"), "proProofRequestBody.version");
+            assertIsNumber(first.Get("unix_ts"), "proProofRequestBody.unix_ts");
+            auto version = first.Get("version").As<Napi::Number>();
+            auto unix_ts = toCppSysMs(first.Get("unix_ts"), "proProofRequestBody.unix_ts");
+
+            assertIsUInt8Array(first.Get("master_privkey"), "proProofRequestBody.master_privkey");
+            assertIsUInt8Array(
+                    first.Get("rotating_privkey"), "proProofRequestBody.rotating_privkey");
+
+            // stack allocate to the buffer view so the ref doesnt get deleted
+            auto master_privkey_napi = first.Get("master_privkey");
+            auto rotating_privkey_napi = first.Get("rotating_privkey");
+            auto master_privkey =
+                    toCppBufferView(master_privkey_napi, "proProofRequestBody.master_privkey");
+            auto rotating_privkey =
+                    toCppBufferView(rotating_privkey_napi, "proProofRequestBody.rotating_privkey");
```

Comment:
I'd advise against this. Copy it with `toCppBuffer`, that will save hours of dev time in the long run. And we are not a copy of 64 bytes away from peak session-desktop performance.